app/components/Authentication/Github/Github.jsx

```diff
@@ -1,248 +1,513 @@
-            .then((resp) => {
-                let policies = _.map(resp.data.policies, (policy) => {
-                    return {
-                        name: policy,
-                        checked: false
-                    }
-                });
-
-                this.setState({
-                    policies: policies
-                })
+    displayItem() {
+        let itemId = this.state.selectedTab;
+        tokenHasCapabilities(['read'], `${this.state.baseVaultPath}/map/${itemId}s/${this.state.selectedItemId}`)
+            .then(() => {
+                callVaultApi('get', `${this.state.baseVaultPath}/map/${itemId}s/${this.state.selectedItemId}`, null, null, null)
+                    .then((resp) => {
+                        resp.data.data.id = itemId;
+                        this.setState({ itemConfig: resp.data.data, openItemDialog: true });
+                    })
+                    .catch(snackBarMessage)
+            })
+            .catch(() => {
+                this.setState({ selectedItemId: '' })
+                snackBarMessage(new Error(`No permissions to display properties for ${itemId} ${this.state.selectedItemId}`));
             })
-            .catch((err) => {
-                console.error(err);
-            });
     }
 
-    submitGithubPolicy() {
-        callVaultApi('post', `auth/github/map/teams/${this.state.teamName}`, null, { value: this.state.checkedPolicies }, null)
-            .then((resp) => {
-
+    createUpdateConfig() {
+        callVaultApi('post', `${this.state.baseVaultPath}/config`, null, this.state.newConfig)
+            .then(() => {
+                snackBarMessage(`Backend ${this.state.baseVaultPath}/config has been updated`);
+                this.setState({ isBackendConfigured: true, config: this.state.newConfig });
             })
-            .catch((err) => {
-                this.setState({ errorMessage: err });
-            });
+            .catch(snackBarMessage);
     }
 
-    getTeamPolicy(teamName) {
-        let uri = `auth/github/map/teams/${this.state.teamName}`;
-        callVaultApi('get', uri, null, null, null)
-            .then((resp) => {
-                let policyNames = _.get(resp, "data.data.value").split(',');
-
-                _.forEach(this.state.policies, (policy) => {
-                    policy.checked = (policyNames.indexOf(policy.name) > -1);
-                });
-                this.setState({
-                    policies: this.state.policies
-                });
+    createUpdateRole(id) {
+        let itemId = this.state.selectedTab;
+        tokenHasCapabilities(['create', 'update'], `${this.state.baseVaultPath}/map/${itemId}s/${id}`)
+            .then(() => {
+                let updateObj = _.clone(this.state.itemConfig);
+                //updateObj.policies = updateObj.policies.join(',');
+                callVaultApi('post', `${this.state.baseVaultPath}/map/${itemId}s/${id}`, null, updateObj)
```

Comment:
CallVaultApi is a shared library, we want to keep it agnostic, so nothing is wrapped. On the class level, a wrapper could definitely be added, though it's not guaranteed the path will be the same.